projects/flimnap/main.cc

```diff
@@ -275,28 +277,26 @@ int main(void) {
         // yaw angle, just use previous state value
         const float yaw_angle = angle::wrap(bicycle.pose().yaw);
 
-        // calculate rider applied torque
 #if defined(FLIMNAP_ZERO_INPUT)
-        const float steer_torque = 0.0f;
-        (void)kistler_torque;
-#else // defined(FLIMNAP_ZERO_INPUT)
-        const float inertia_torque = -handlebar_model.feedback_torque(bicycle.observer().state());
-        const float steer_torque = kistler_torque - inertia_torque;
+        // Need to calculate what kistler torque _should_ be:
+        // 0 = kistler_torque + upper_inertia_torque
+        const float kistler_torque = -bicycle.inertia_upper_virtual().torque(bicycle.observer().state());
 #endif // defined(FLIMNAP_ZERO_INPUT)
 
         // simulate bicycle
         bicycle.set_v(fixed_velocity);
-        bicycle.update_dynamics(roll_torque, steer_torque, yaw_angle, steer_angle, rear_wheel_angle);
+        bicycle.update_dynamics(roll_torque, kistler_torque, yaw_angle, steer_angle, rear_wheel_angle);
 
         // generate handlebar torque output
-        const dacsample_t handlebar_torque_dac = set_handlebar_torque(bicycle.handlebar_feedback_torque());
+        const float feedback_torque = bicycle.handlebar_feedback_torque();
+        const dacsample_t handlebar_torque_dac = set_handlebar_torque(feedback_torque);
         chTMStopMeasurementX(&computation_time_measurement);
 
         // prepare message for transmission
         msg = msg_init;
         msg.timestamp = starttime;
         message::set_bicycle_input(&msg.input,
-                (model_t::input_t() << roll_torque, steer_torque).finished());
+                (model_t::input_t() << roll_torque, kistler_torque).finished());
```

Comment:
These values are the input to the Whipple model. As they are calculated in the sim::Bicycle class, we probably need to add a member variable to store the most recent calculation and an accessor.